components/microsoft_onedrive/sources/common/base.mjs

```diff
@@ -0,0 +1,261 @@
+import onedrive from "../../microsoft_onedrive.app.mjs";
+import { toSingleLineString } from "./utils.mjs";
+
+// Defaulting to 15 days. The maximum allowed expiration time is 30 days,
+// according to their API response message: "Subscription expiration can only
+// be 43200 minutes in the future".
+//
+// More information can be found in the official API docs:
+// https://docs.microsoft.com/en-us/onedrive/developer/rest-api/concepts/using-webhooks?view=odsp-graph-online#expiration
+const WEBHOOK_SUBSCRIPTION_RENEWAL_SECONDS = 43200 * 60 / 2;
+
+export const props = {
+  onedrive,
+  db: "$.service.db",
+  http: {
```

Comment:
Can we put `props`, `hooks`, `methods`, and `run` within an `export default`? That makes it more straightforward that all of `base` is being inherited by any components that use it.